test/test_report.py

```diff
@@ -89,3 +91,90 @@ def test_averages_three_samples() -> None:
     averages.update(sample2)
     averages.update(sample2)
     assert averages.pcpu == (stat0.pcpu + (2 * stat1.pcpu)) / 3
+
+
+def test_process_stats_green() -> None:
+    # Assert does not raise
+    ProcessStats(
+        pcpu=1.0,
+        pmem=1.1,
+        rss=1024,
+        vsz=1025,
+        timestamp=datetime.now().astimezone().isoformat(),
+    )
+
+
+def test_process_stats_handle_ints() -> None:
+    # Assert does not raise
+    ProcessStats(
+        pcpu=1,
+        pmem=1,
+        rss=1024,
+        vsz=1025,
+        timestamp=datetime.now().astimezone().isoformat(),
+    )
+
+
+def test_process_stats_incorrect_pcpu_type() -> None:
+    with pytest.raises(TypeError):
+        ProcessStats(
+            pcpu="oopsie",  # type: ignore[arg-type]
+            pmem=1.1,
+            rss=1024,
+            vsz=1025,
+            timestamp=datetime.now().astimezone().isoformat(),
+        )
+
+
+def test_process_stats_incorrect_pmem_type() -> None:
+    with pytest.raises(TypeError):
+        ProcessStats(
+            pcpu=1.1,
+            pmem="oopsie",  # type: ignore[arg-type]
+            rss=1024,
+            vsz=1025,
+            timestamp=datetime.now().astimezone().isoformat(),
+        )
+
+
+def test_process_stats_incorrect_rss_type() -> None:
+    with pytest.raises(TypeError):
+        ProcessStats(
+            pcpu=1.1,
+            pmem=1.0,
+            rss="oopsie",  # type: ignore[arg-type]
+            vsz=1025,
+            timestamp=datetime.now().astimezone().isoformat(),
+        )
+
+
+def test_process_stats_incorrect_vsz_type() -> None:
+    with pytest.raises(TypeError):
+        ProcessStats(
+            pcpu=1.1,
+            pmem=1.0,
+            rss=1025,
+            vsz="oopsie",  # type: ignore[arg-type]
+            timestamp=datetime.now().astimezone().isoformat(),
+        )
+
+
+def test_process_stats_incorrect_ts_type() -> None:
+    with pytest.raises(TypeError):
+        ProcessStats(
+            pcpu=1.1,
+            pmem=1.0,
+            rss=1025,
+            vsz=1024,
+            timestamp=1  # type: ignore[arg-type]
+        )
```

Comment:
duplication makes it hard(er) to see the critical difference (even though it is in the name and even in type annotation comment).  IMHO would have been much better done via `@pytest.mark.parametrize` having some common `kwargs` dict which you `.update` with the specific parameter (specific `attribute=value`) and have a generic construct. Makes then really easy to see what that particular test is doing, that it is just 1 test and different parametrizations.